src/dra/6D59C.c

```diff
@@ -71,7 +71,68 @@ void func_8010D59C(void) {
     }
 }
 
-INCLUDE_ASM("dra/nonmatchings/6D59C", func_8010D800);
+u8 g_D_800ACF2C[] = {4, 4, 4, 4, 6, 6, 6, 6, 8, 8, 16, 16, 16, 16, 16, 16};
+u8 g_D_800ACF3C[] = {8,  12, 16, 20, 24, 28, 32, 32,
+                     32, 32, 32, 32, 32, 32, 32, 32};
+
+void func_8010D800(void) {
+    byte pad[0x28];
+    Unkstruct_800ECBF8_1* current_thing;
+    Primitive* prim;
+    s32 entNum;
+    s32 i;
+    u8 temp_t0;
+    u8 temp_t1;
+    u8 temp_t2;
+
+    i = 0;
+    prim = &g_PrimBuf[g_Entities[1].primIndex];
+    temp_t2 = g_Entities[1].ext.ent1.unk1;
+    temp_t1 = g_D_800ACF2C[g_Entities[1].ext.ent1.unk2];
+    temp_t0 = g_D_800ACF3C[g_Entities[1].ext.ent1.unk2];
+
+    current_thing = &D_80097D1C[1];
+    for (i = 0; i < 6; prim = prim->next, i++) {
+        if (temp_t0 < prim->r0) {
+            prim->r0 -= temp_t1;
```

Comment:
`prim->r0` seems to be opacity. I'm wondering if there's other meanings to these values we haven't figured out.